build.sh

```diff
@@ -40,10 +40,10 @@ DOTNET="$(pwd)/cli/dotnet"
 $DOTNET --info
 
 # Get CLI Branches for testing
-echo "dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
+echo "$DOTNET msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
```

Comment:
`$DOTNET` points to `$RepoRoot/cli/dotnet`, which is intended for use by two of our test projects (Dotnet.Integration.Tests and MSBuild.Integration.Tests), but we prefer production/stable tooling to compile our repo and run our other scripts.

Is there some reason I can't think of why we wouldn't want to use the system-installed `dotnet`?

Reply:
By the looks of it, `eng/source-build/build.sh` doesn't use `build.sh`, so I propose reverting all changes to this file.

Reply:
Reverted this as suggested, thanks.